submissions/usides/memory-game/js/main.js

```diff
@@ -0,0 +1,123 @@
+const gameBoard = document.getElementById('gameBoard');
+const newGameBtn = document.getElementById('newGameBtn');
+const mistCount = document.getElementById('mistCount');
+const secCount = document.getElementById('secCount');
+const meowSound = new Audio('./sounds/meow.mp3');
+const lockSound = new Audio('./sounds/lock.mp3');
+
+const PICS = [
+  'crim_01.jpg',
+  'crim_02.jpg',
+  'crim_03.jpg',
+  'crim_04.jpg',
+  'crim_05.jpg',
+  'crim_06.jpg',
+];
+
+const pairsMax = PICS.length;
+let pairsOpen = 0;
+let firstOpen = null;
+let secondOpen = null;
+let secCountTimer = null;
+
+gameBoard.addEventListener('click', cardClick);
+
+newGameBtn.addEventListener('click', function () {
+  this.classList.remove('visible');
+  newGame();
+});
+
+const cards = PICS.flatMap((pict) => {
+  const newElem1 = document.createElement('div');
+  const newElem2 = document.createElement('div');
+  newElem1.classList.add('card');
+  newElem2.classList.add('card');
+  newElem1.setAttribute('data-card', pict.match(/[^.]+/gi)[0]);
+  newElem2.setAttribute('data-card', pict.match(/[^.]+/gi)[0]);
+  const template = `<div class="back"><img src="./img/logo.svg" alt="Logo"></div><div class="front"><img src="./img/${pict}" alt="Cat Picture">
+    <img class='grid' src="./img/grid.png" alt="Grid"></div>`;
+  newElem1.innerHTML = template;
+  newElem2.innerHTML = template;
+  return [newElem1, newElem2];
```

Comment:
why you double your code ?
you can do just one element then return array which contains two same elements in flatMap
```
const newElem = document.createElement('div');
...
return [newElem, newElem];
```

Reply:
I can't return an array that contains two **same** elements. We need an array with 12 **unique** elements but if I do it, like you propose, I will have only 6 unique elements in the array and 6 copies(reference copies). When I do append operation I append the same element twice and during second append operation browser just make a **relocation** of this element (not insertion of the new one). As a result we have only 6 cards on the field, but we need 12.

Reply:
Okay, i understand you. My bad. I really forgot that elements will have same pointer.
Small example =)
```
const arr = [1,2,3,4];

const cards = arr.flatMap(n => {
    const card = document.createElement('div');
    card.innerText  = n;
    return [card, card];
})

console.log(cards[0].innerText, cards[1].innerText) // 1 1
cards[0].innerText = 'haha'; // will change for both cards because they refer to the same object
console.log(cards[0].innerText, cards[1].innerText) // haha haha
```

To prevent this you can do like this.

```
const arr = [1,2,3,4];

const createCard = number => {
    const card = document.createElement('div');
    card.innerText = number;
    return card;
}

const cards = arr.flatMap(n => [createCard(n), createCard(n)])
```

Reply:
@usides

Reply:
Done.